cli/src/merge_tools/mod.rs

```diff
@@ -158,22 +159,19 @@ pub fn get_external_tool_config(
     settings: &UserSettings,
     name: &str,
 ) -> Result<Option<ExternalMergeTool>, ConfigError> {
-    const TABLE_KEY: &str = "merge-tools";
-    let tools_table = settings.get_table(TABLE_KEY)?;
-    if let Some(v) = tools_table.get(name) {
-        let mut result: ExternalMergeTool = v
-            .clone()
-            .try_deserialize()
-            // add config key, deserialize error is otherwise unclear
-            .map_err(|e| ConfigError::Message(format!("{TABLE_KEY}.{name}: {e}")))?;
-
-        if result.program.is_empty() {
-            result.program.clone_from(&name.to_string());
-        };
-        Ok(Some(result))
-    } else {
-        Ok(None)
-    }
+    let full_name = ConfigNamePathBuf::from_iter(["merge-tools", name]);
+    let Some(mut tool) = settings
+        .get::<ExternalMergeTool>(&full_name)
```

Comment:
From commit message:
> cli: leverage new settings.get(dotted_name) in merge-tools lookup

We don't seem to use the dotted_name form here. Is the commit message stale?

Reply:
Fixed as s/dotted_name/path/. Maybe I just called the name path type as dotted_name.